adr/ADR-37.md

```diff
@@ -27,37 +27,75 @@ performance.
 
 ## Design
 
-JetStream simplification revolves around the concept of the `Consumer`. The
-consumer definition managed by the server is created by tools (such as nats cli)
-or code. A `Consumer` can be created or added as per the client language
-implementation, typically by providing a stream name and some consumer options.
-
-Clients wanting to consume messages from the consumer, simply locate the
-`Consumer`, which then provides functionality for processing, or consuming
-messages. The representation of the `Consumer` is retrieved, and provides
-operations to `fetch` and `consume` messages.
+The JetStream API consists of three main components: JetStreamContext, Streams,
+and Consumers.
+
+### JetStream Context
+
+The JetStream Context is mainly responsible for managing streams. It serves as
+the entry point for creating, configuring, and controlling the streams.
+JetStreamContext should also expose methods co manage consumers directly,
+bypassing the need to get/create a stream.
+
+Example set of methods on JetStreamContext:
+
+- operations on streams:
+  - `addStream(streamConfig)`
+  - `updateStream(streamConfig)`
+  - `getStream(streamName)`
+  - `deleteStream(streamName)`
+  - `listStreams()`
+- operations on consumers:
+  - `addConsumer(streamName, consumerConfig)`
+  - `getConsumer(streamName, consumerName)`
+  - `deleteConsumer(streamName, consumerName)`
+- `accountInfo()`
+
+### Streams
```

Comment:
Does anyone feel like this just ends up hiding the server meaning of stream. Developers are already used the the word context. A context makes sense to me since it represents a stream and operations that can be done with and around stream.